zkevm-circuits/src/witness/block.rs

```diff
@@ -4,13 +4,17 @@ use bus_mapping::circuit_input_builder::{self, CircuitsParams, CopyEvent, ExpEve
 use eth_types::{Address, Field, ToLittleEndian, ToScalar, Word};
 use halo2_proofs::halo2curves::bn256::Fr;
 use itertools::Itertools;
-
-use crate::{evm_circuit::util::RandomLinearCombination, table::BlockContextFieldTag};
+use crate::{
+    evm_circuit::util::RandomLinearCombination,
+    table::BlockContextFieldTag,
+    evm_circuit::step::ExecutionState,
+};
 
 use super::{step::step_convert, tx::tx_convert, Bytecode, ExecStep, RwMap, Transaction};
 
-/// Block is the struct used by all circuits, which constains all the needed
+/// Block is the struct used by all circuits, which constraints all the needed
 /// data for witness generation.
+/// TODO: figure out better way to deal with potential ExecutionState mismatch when invoke default trait
```

Comment:
Since `Block::default()` is no longer valid, I think it's reasonable to remove the `Default` derive and use `Block::new` to create a valid empty block.

Another way is to call `Block::new()` inside `Block::default`, tho it might be confusing sometimes.